pd/src/components/staking.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use penumbra_chain::params::ChainParams;
+use penumbra_stake::Epoch;
+use penumbra_transaction::Transaction;
+use tendermint::abci;
+
+use super::{Component, Overlay};
+use crate::{components::validator_set::BlockChanges, genesis, WriteOverlayExt};
```

Comment:
I think it'd be better to firewall off this code from the previous `validator_set` code, and just copy things over as needed, so that when we're done, we can cleanly delete the old code.  In this case, for instance, I think the design of the `WriteOverlay` will probably mean we don't need the `BlockChanges` at all.